### ISSUE TITLE
Problem w/ branch len estimate with closely related leaves

### ISSUE THREAD

**Issue by phiweger** [neherlab/treetime#134]
In the TreeTime .nexus output I get a huge negative branch len followed by another large on for the corresponding leaves:

```
...
((7ab5cd3e-524b-4f3e-9951-04d783bcef78:28113.25709,5abe8078-fdb6-4e90-9075-314bc4238f48:28113.15326)NODE_0000024:-28112.91947,(dd518f5c-d48a-464d-bff3-4becb51ae5d5:0.00000,0e2e43b1-45fc-4a32-b584-d4db7b91e86b:0.00000)
...
```

Is this a bug or some numerical instability? How could I avoid this?

Thanks a lot!

**Comment by phiweger** [comment]
Further testing gives me the impression that (1) this does not always occur given the same input and (2) only occured when I add the --confidence flag to treetime.

**Comment by rneher** [comment]
it this run a tree with four leaves with some identical dates/branch lengths? then it is likely a numerical instability when trying to invert a singular matrix.

**Comment by phiweger** [comment]
Yes, this is a larger tree (20+ leaves) but 3 of them are identical in their SNV alignment, but the dates are different. Is there a way around this instability, besides manually clipping the corresponding branch values to 0? The dates should help resolve polytomies, right?

**Comment by rneher** [comment]
could you send me these data. I can't quite explain why this might happen and it would be good to fix.

**Comment by phiweger** [comment]
which data do you need? the alignment, dates, undated tree -- anything else?

**Comment by rneher** [comment]
yes, those are what I would need.

**Comment by ktmeaton** [comment]
I think I might be having a similar error (if not I can open a new issue). When estimating date confidences using the marginal likelihood, some nodes will sporadically have very large intervals:

![image](https://user-images.githubusercontent.com/14981272/113437439-92ba2080-93b4-11eb-9358-72b7ee112d24.png)

Rather than having intervals in the range of 100s of years, these nodes have confidence intervals of +100,000 years. These large intervals are somewhat random, in that rerunning the analyses moves them around. Any thoughts on why this might be occurring and if there's a solution?

**Comment by rneher** [comment]
yes, this looks like there is a problem. My hunch is that there is some numerical accuracy problem.

**Comment by ktmeaton** [comment]
I was thinking numerical accuracy too. This is a large phylogeny with many small branches (1e-8). Would there be any value in rescaling the branch lengths before (ex. multiply them all by 1e4)?

**Comment by rneher** [comment]
I suppose this is a large genome? Does this use a SNP only alignment? Or a vcf file? TreeTime carries around an internal scale that is `one_mutation = 1/L` (L being the length of the genome). One could just try to trick it in assuming the genome is shorter. But I am not sure I understand your application well enough.

**Comment by m-a-martin** [comment]
> I think I might be having a similar error (if not I can open a new issue). When estimating date confidences using the marginal likelihood, some nodes will sporadically have very large intervals:
> 
> ![image](https://user-images.githubusercontent.com/14981272/113437439-92ba2080-93b4-11eb-9358-72b7ee112d24.png)
> 
> Rather than having intervals in the range of 100s of years, these nodes have confidence intervals of +100,000 years. These large intervals are somewhat random, in that rerunning the analyses moves them around. Any thoughts on why this might be occurring and if there's a solution?

I am having this same (or a similar) issue on a SARS-CoV-2 dataset with roughly 5000 sequences using the flags: 

```
-tree ml_clean.nwk --dates clean_metadata.tsv --aln aln_clean.fasta --clock-filter 4 --reroot EPI_ISL_402125 --covariation --coalescent skyline --clock-rate 0.001 --clock-std-dev 0.0005 --branch-length-mode joint --confidence --keep-polytomies
```

I'm using a full alignment. The problem is random and rerunning on the same dataset can generate reasonable confidence intervals, but it happens often enough that it is an issue. Using TreeTime v. 0.80 on Python v3.9. I've attached the treetime output as well as the ML tree and a list of accession numbers (can't share alignment because GISAID data). 

[for_github.zip](https://github.com/neherlab/treetime/files/6694214/for_github.zip)
